src/Compiler/Microsoft.AspNetCore.Razor.Language/test/Legacy/HtmlBlockTest.cs

```diff
@@ -262,6 +262,39 @@ public void DoesNotConsiderPsuedoTagWithinMarkupBlock()
         ParseDocumentTest("@{<foo><text><bar></bar></foo>}");
     }
 
+    [Fact, WorkItem("https://github.com/dotnet/razor/issues/10180")]
+    public void TextAfterCodeBlockInMarkupTransition_01()
+    {
+        ParseDocumentTest("""
+            @{
+                @:@{ <i>x y z </i> }
```

Comment:
Is it worth adding some verisons of these that split the code block across multiple lines?